### ISSUE TITLE
Silencing clippy warnings

### ISSUE THREAD

**Issue by arthmis** [microsoft/windows-rs#248]
When building my project using the `build!()` macro the winrt.rs file is generated and put into the `OUT_DIR`. Now when I include this in `main.rs` with `include!(concat!(env!("OUT_DIR"), "/winrt.rs"))` and rust-analyzer uses clippy for cargo check, I get a huge amount of warnings. Most of them are about unreadable literals. 

Is it possible to have `#[allow(clippy::all)]` at the top of the generated `winrt.rs` file so the warnings aren't included in rust-analyzer clippy output for my project's source files?

**Comment by kennykerr** [comment]
Sorry for the delay @arthmis - I'll certainly take a look at making the build "clippy clean". 😉

**Comment by gaelog** [comment]
I see good progress being made on fixing the lints on a case-per-case basis. New linter warnings will likely be introduced, posing valid concerns. Disabling them inside of the generated windows.rs is a quick fix, but would only mask the issue.

It's possible to disable the warnings on the parent module when including the bindings:
```rust
#[allow(dead_code, clippy::all)]
mod bindings {
    ::windows::include_bindings!();
}
```
This should get rid of any warnings (tested with [windows-docs-rs](https://github.com/microsoft/windows-docs-rs)).
It's not the cleanest, but since the code is technically generated inside the client crate, that much responsability seems fair to me?
Also it is much faster to toggle the `dead_code` here (e.g. when cleaning your `build.rs`) instead of rebuilding. Introducing crate features to toggle clippy warnings seems like an overkill.

**Comment by kennykerr** [comment]
Fixed as of #637
